coredhcp.go

```diff
@@ -104,11 +104,21 @@ func (s *Server) LoadPlugins(conf *config.Config) ([]*plugins.Plugin, []*plugins
 // It will not reply if the resulting response is `nil`.
 func (s *Server) MainHandler6(conn net.PacketConn, peer net.Addr, req dhcpv6.DHCPv6) {
 	var (
-		resp dhcpv6.DHCPv6
-		stop bool
-		err  error
+		resp, originalReq dhcpv6.DHCPv6
+		stop              bool
+		err               error
 	)
 
+	// decapsulate the relay message
```

Comment:
How about doing `msg, err := packet.GetInnerMessage()` here and you can avoid the `if req.IsRelay()`.

And then using the `msg` variable everywhere and you can avoid the type assertion of lines 125, 133 and 151.

Reply:
thanks, I'll look into this :)

Reply:
that works, but I still need to save the original message for the re-encapsulation below

Reply:
The original message is in function parameter, so it is still there.

Reply:
done